pkg/pbq/pbq.go

```diff
@@ -0,0 +1,101 @@
+package pbq
+
+import (
+	"context"
+	"errors"
+	"github.com/numaproj/numaflow/pkg/isb"
+	"github.com/numaproj/numaflow/pkg/pbq/store"
+	"go.uber.org/zap"
+)
+
+var COBErr error = errors.New("error while writing to pbq, pbq is closed")
+
+type PBQ struct {
+	store       store.Store
+	output      chan *isb.Message
+	cob         bool // cob to avoid panic in case writes happen after close of book
+	partitionID string
+	options     *Options
+	manager     *Manager
+	log         *zap.SugaredLogger
+}
+
+// Write writes message to pbq and persistent store
+// We don't need a context here as this is invoked for every message.
+func (p *PBQ) Write(ctx context.Context, message *isb.Message) (writeErr error) {
```

Comment:
```suggestion
func (p *PBQ) Write(ctx context.Context, message *isb.Message) (error) {
```

usually this is done to manipulate the return result via `defer`

Reply:
yeah, agreed